presto-bigquery/src/main/java/com/facebook/presto/plugin/bigquery/BigQueryColumnHandle.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.bigquery;
+
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.spi.ColumnMetadata;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static com.facebook.presto.plugin.bigquery.BigQueryType.Adaptor;
+import static com.google.cloud.bigquery.Field.Mode;
+import static com.google.cloud.bigquery.Field.Mode.NULLABLE;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class BigQueryColumnHandle
+        implements ColumnHandle, Adaptor
+{
+    private final String name;
+    private final BigQueryType bigQueryType;
+    private final Mode mode;
+    private final List<BigQueryColumnHandle> subColumns;
+    private final String description;
+
+    @JsonCreator
+    public BigQueryColumnHandle(
+            @JsonProperty("name") String name,
+            @JsonProperty("bigQueryType") BigQueryType bigQueryType,
+            @JsonProperty("mode") Mode mode,
+            @JsonProperty("subColumns") List<BigQueryColumnHandle> subColumns,
+            @JsonProperty("description") String description)
+    {
+        this.name = requireNonNull(name, "column name cannot be null");
+        this.bigQueryType = requireNonNull(bigQueryType, () -> format("column type cannot be null for column [%s]", name));
+        this.mode = requireNonNull(mode, "Field mode cannot be null");
+        this.subColumns = ImmutableList.copyOf(requireNonNull(subColumns, "subColumns is null"));
+        this.description = description;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;description&#39; (index: 4 | type: String). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;subColumns&#39;, &#39;mode&#39;, &#39;name&#39;, &#39;bigQueryType&#39;. The same parameter type is validated here for example: presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java:429. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).